toolchain/check/check.cpp

```diff
@@ -444,7 +444,7 @@ class DeferredDefinitionWorklist {
     std::optional<DeclNameStack::SuspendedName> suspended_name;
     // Whether this scope is itself within an outer deferred definition scope.
     // If so, we'll delay processing its contents until we reach the end of the
-    // enclosing scope. For example:
+    // parent scope. For example:
```

Comment:
We mean the lexically-enclosing scope rather than necessarily the parent scope. For example:
```
class A {
  class B {
    class C;
  }
  class B.C {
    fn F() {}
  }
}
```
The scope of `A.B.C` is within an outer deferred definition scope -- namely, the scope of `A` -- which is not the parent scope (which would be `A.B`). How about:
```suggestion
    // outer scope. For example:
```